sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/HDFSMetadataLog.scala

```diff
@@ -63,8 +63,39 @@ class HDFSMetadataLog[T <: AnyRef : ClassTag](sparkSession: SparkSession, path:
   val metadataPath = new Path(path)
   protected val fileManager = createFileManager()
 
-  if (!fileManager.exists(metadataPath)) {
-    fileManager.mkdirs(metadataPath)
+  runUninterruptiblyIfLocal {
+    if (!fileManager.exists(metadataPath)) {
+      fileManager.mkdirs(metadataPath)
+    }
+  }
+
+  private def runUninterruptiblyIfLocal[T](body: => T): T = {
+    if (fileManager.isLocalFileSystem) {
+      Thread.currentThread match {
+        case ut: UninterruptibleThread =>
+          // When using a local file system, some file system APIs like "create" or "mkdirs" must be
```

Comment:
Fixed the comment. I added it in https://github.com/apache/spark/commit/88c43f4fb5ea042a119819c11a5cdbe225095c54 but it was wrong. We don't need to use `runUninterruptibly ` to workaround `HADOOP-14084`. The root cause is `HADOOP-10622`.